code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -905,77 +912,122 @@ pub mod pallet {
 			(fnft_collection_id, fnft_instance_id): Self::PositionId,
 			amount: Self::Balance,
 			keep_alive: bool,
-		) -> Result<Self::PositionId, DispatchError> {
-			let mut stake = Stakes::<T>::get(fnft_collection_id, fnft_instance_id)
-				.ok_or(Error::<T>::StakeNotFound)?;
-			let mut rewards_pool = RewardPools::<T>::try_get(stake.reward_pool_id)
-				.map_err(|_| Error::<T>::RewardsPoolNotFound)?;
+		) -> DispatchResult {
+			Stakes::<T>::try_mutate(fnft_collection_id, fnft_instance_id, |maybe_stake| {
+				let stake = maybe_stake.as_mut().ok_or(Error::<T>::StakeNotFound)?;
 
-			let reward_multiplier = FixedU64::one().try_into_validated().expect(">= 1");
+				RewardPools::<T>::try_mutate(stake.reward_pool_id, |maybe_rewards_pool| {
+					let rewards_pool =
+						maybe_rewards_pool.as_mut().ok_or(Error::<T>::RewardsPoolNotFound)?;
 
-			ensure!(
-				matches!(
-					T::Assets::can_withdraw(rewards_pool.asset_id, who, amount),
-					WithdrawConsequence::Success
-				),
-				Error::<T>::NotEnoughAssets
-			);
+					ensure!(
+						matches!(
+							T::Assets::can_withdraw(rewards_pool.asset_id, who, amount),
+							WithdrawConsequence::Success
+						),
+						Error::<T>::NotEnoughAssets
+					);
 
-			let awarded_shares = Self::boosted_amount(reward_multiplier, amount)?;
+					let new_shares = Self::boosted_amount(
+						// SAFETY: The duration preset on an existing stake should be valid in the
+						// pool since it's currently not possible to modify the presets after pool
+						// creation.
+						rewards_pool
```

Comment:
In-lining this in the function decreases the readability at least in review because it's not clear what value is being passed.

Reply:
https://github.com/ComposableFi/composable/commit/dda75e674d708b76807572b440ffc30306e49f41